language_service/src/test_utils.rs

```diff
@@ -66,10 +69,99 @@ pub(crate) fn compile_with_fake_stdlib(source_name: &str, source_contents: &str)
         PackageType::Exe,
         TargetProfile::Full,
     );
+
+    let package_id = package_store.insert(unit);
+
+    Compilation {
+        package_store,
+        user: package_id,
+        kind: CompilationKind::OpenDocument,
+        errors,
+    }
+}
+
+pub(crate) fn compile_notebook_with_fake_stdlib_and_markers(
+    cells_with_markers: &[(&str, &str)],
+) -> (Compilation, String, u32, Option<String>, Vec<u32>) {
+    let (mut cell_uri, mut offset, mut target_cell_uri, mut target_offsets) =
+        (None, None, None, Vec::new());
+    let cells = cells_with_markers
+        .iter()
+        .map(|c| {
+            let (source, cursor_offsets, targets) = get_source_and_marker_offsets(c.1);
+            if !cursor_offsets.is_empty() {
+                assert!(
+                    cell_uri.replace(c.0).is_none(),
+                    "only one cell can have a cursor marker"
+                );
+                assert!(
+                    offset.replace(cursor_offsets[0]).is_none(),
+                    "only one cell can have a cursor marker"
+                );
+            }
+            if !targets.is_empty() {
+                assert!(
+                    target_cell_uri.replace(c.0).is_none(),
+                    "only one cell can have a target marker"
```

Comment:
Why can only one cell contain target markers? I can imagine a rename or find-all-references operation that targets things in multiple cells.

Reply:
Well, this was convenient right now since other unit tests don't use target markers. You could totally update those unit tests to use target markers :) . Then we can relax this restriction.

Reply:
I've updated the other unit tests in the language server to use target markers where possible.